mlflow/cli/eval_utils.py

```diff
@@ -0,0 +1,188 @@
+        raise click.UsageError("No valid scorers specified")
+
+    return resolved_scorers
+
+
+def extract_assessments_from_results(
+    results_df: pd.DataFrame, evaluation_run_id: str
+) -> list[dict[str, Any]]:
+    """
+    Extract assessments from evaluation results DataFrame.
+
+    The evaluate() function returns results with a DataFrame that contains
+    an 'assessments' column. Each row has a list of assessment dictionaries
+    with metadata including 'mlflow.assessment.sourceRunId' that we use to
+    filter assessments from this specific evaluation run.
+
+    Args:
+        results_df: DataFrame from evaluate() results containing assessments column
+        evaluation_run_id: The MLflow run ID from the evaluation that generated the assessments
+
+    Returns:
+        List of dictionaries with trace_id and assessments
+    """
+    output_data = []
+
+    for _, row in results_df.iterrows():
+        trace_id = row["trace_id"]
+        trace_result = {"trace_id": trace_id, "assessments": []}
+
+        assessments_list = row.get("assessments", [])
+
+        if assessments_list:
+            for assessment_dict in assessments_list:
+                # Check if this assessment is from our evaluation run
+                metadata = assessment_dict.get("metadata", {})
+                source_run_id = metadata.get("mlflow.assessment.sourceRunId")
+
+                if source_run_id == evaluation_run_id:
+                    result_dict = {
+                        "assessment_name": assessment_dict.get("assessment_name"),
+                        "result": None,
+                        "rationale": None,
+                    }
+
+                    # Extract feedback value
+                    feedback = assessment_dict.get("feedback")
+                    if feedback and isinstance(feedback, dict):
+                        result_dict["result"] = feedback.get("value")
+
+                    # Extract rationale
+                    rationale = assessment_dict.get("rationale")
+                    if rationale:
+                        result_dict["rationale"] = rationale
+
+                    # Check for errors (if assessment has error info)
+                    error = assessment_dict.get("error")
+                    if error:
+                        result_dict["error"] = str(error)
+
+                    trace_result["assessments"].append(result_dict)
+
+        # If no assessments were found for this trace, add error markers
+        if not trace_result["assessments"]:
+            trace_result["assessments"].append(
+                {
+                    "assessment_name": "N/A",
+                    "result": None,
+                    "rationale": None,
+                    "error": "No assessments found on trace",
+                }
+            )
+
+        output_data.append(trace_result)
+
+    return output_data
+
+
+def format_table_output(output_data: list[dict[str, Any]]) -> tuple[list[str], list[list[str]]]:
```

Comment:
Instead of returning a tuple of lists / list of lists, can we define an internal dataclass and return that instead?